### PR TITLE
[FIX] check if the key is present in the result before sort result on it

### PR DESCRIPTION
**Pull request by vrenaville**
check if the key is present in the result before sort result on it

### CODE REVIEW

File: budget/budget_line.py [hunk 1]
```diff
@@ -386,5 +386,6 @@ def read_group(self, cr, uid, domain, fields, groupby, offset=0,
                 reverse = (order[0][1] == 'DESC')
             getter = [x[0] for x in order if x[0]]
             if getter:
-                res = sorted(res, key=itemgetter(*getter), reverse=reverse)
+                if getter[0] in res[0].keys():
```
Comment (pedrobaeza) [inline budget/budget_line.py]: There can be more than one key, so you must check all of them. You can use something like:

```
if all([x in res[x].keys() for x in getter]):
```

**Comment by vrenaville** [comment]
Thanks Pedro, 
I have forgot to push my last commit, but I found your solution cleaner, so I have improve my PR

**Comment by pedrobaeza** [comment]
Maybe you can build getter taking this into account instead of checking afterwards, and then you will get the result sorted only by the existing keys instead of none.

Regards.

**Comment by vrenaville** [comment]
@nbessi  Nicolas can you please review it, and make a comment

**Comment by bwrsandman** [comment]
You're iterating over the data twice, this doesn't seem very efficient.
I would suggest a try-except block from the idiom "It's easier to ask for forgiveness than for permission"

**Pull request closed without merging**